examples/usb-serial/src/main.rs

```diff
@@ -0,0 +1,75 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+#![feature(used_with_arg)]
+
+use riot_rs::{
+    debug::log::info,
+    embassy::{
+        embassy_usb, make_static,
+        usb::{UsbBuilderHook, UsbDriver},
+    },
+};
+
+use embassy_usb::{
+    class::cdc_acm::{CdcAcmClass, State},
+    driver::EndpointError,
+};
+
+#[riot_rs::config(usb)]
+fn usb_config() -> embassy_usb::Config<'static> {
+    let mut config = embassy_usb::Config::new(0xc0de, 0xcafe);
+    config.manufacturer = Some("RIOT-rs");
+    config.product = Some("USB serial example");
+    config.serial_number = Some("12345678");
+    config.max_power = 100;
+    config.max_packet_size_0 = 64;
+
+    // Required for Windows support.
+    config.composite_with_iads = true;
+    config.device_class = 0xEF;
+    config.device_sub_class = 0x02;
+    config.device_protocol = 0x01;
```

Comment:
This implies that we're talking AT commands, which is not only incorrect, but it is also an invitation for NetworkManager and similar to start probing the apparent modem, sending garbage before the device is made available to the user. RIOT OS picks 0 (USB_CDC_PROTOCOL_NONE); 0xFF (vendor-specific) might be a sensible choice as well.

Reply:
This is copied from upstream. the subdescriptor is configures with `USB_CDC_PROTOCOL_NONE` iirc: https://github.com/kaspar030/embassy/blob/584940fc7ea02fe184f3ae346ecaa3d00008c95e/embassy-usb/src/class/cdc_acm.rs#L219